### PR TITLE
[chaotic-good] Clean up unused requested_read_args in PromiseEndpoint::ReadCallback.

### PR DESCRIPTION
**Pull request by nanahpang**
This is a clean up PR for discussion in #33257.

### CODE REVIEW

**Review by Vignesh2208 — APPROVED** [review]

File: src/core/lib/transport/promise_endpoint.cc [hunk 1]
```diff
@@ -88,18 +85,17 @@ void PromiseEndpoint::ReadCallback(
     if (read_buffer_.Length() < num_bytes_requested) {
       // A further read is needed.
       // Set read args with number of bytes needed as hint.
-      requested_read_args = {
-          static_cast<int64_t>(num_bytes_requested - read_buffer_.Length())};
+      grpc_event_engine::experimental::EventEngine::Endpoint::ReadArgs
+          read_args;
+      read_args.read_hint_bytes =
```
Comment (ralphchung) [inline src/core/lib/transport/promise_endpoint.cc:90]: Minor: Could it initialize in the same line?

File: src/core/lib/transport/promise_endpoint.h [hunk 2]
```diff
@@ -158,16 +157,16 @@ class PromiseEndpoint {
     if (read_buffer_.Length() < num_bytes) {
       lock.Release();
       // Set read args with num_bytes as hint.
-      const struct grpc_event_engine::experimental::EventEngine::Endpoint::
-          ReadArgs read_args = {static_cast<int64_t>(num_bytes)};
+      grpc_event_engine::experimental::EventEngine::Endpoint::ReadArgs
+          read_args;
+      read_args.read_hint_bytes = num_bytes;
```
Comment (ralphchung) [inline src/core/lib/transport/promise_endpoint.h:162]: Same as above.

**Review by ralphchung — APPROVED** [review]

File: src/core/lib/transport/promise_endpoint.cc (hunk 1 above)
Comment (nanahpang) [reply src/core/lib/transport/promise_endpoint.cc]: Done. Thanks for suggestion.

File: src/core/lib/transport/promise_endpoint.h (hunk 2 above)
Comment (nanahpang) [reply src/core/lib/transport/promise_endpoint.h]: Done.

**Pull request merged**